### PR TITLE
Add vapor server and shared lib

### PR DESCRIPTION
**Pull request by sivayogasubramanian**
Open package.swift in Xcode and click Run (My Mac).

You should also be able to import the shared lib on both the server package as well as the main project.

### CODE REVIEW

File: AnnotatoBackend/Package.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// swift-tools-version:5.5
+import PackageDescription
+
+let package = Package(
+    name: "AnnotatoBackend",
+    platforms: [
+       .macOS(.v12)
```
Comment (ganhongyao) [inline AnnotatoBackend/Package.swift]: Should we use the v10_15 version of Vapor to be more flexible? I haven't updated to macos 12 and will not be able to run the app

Comment (ganhongyao) [reply AnnotatoBackend/Package.swift]: To use the v10_15 version, use the flag `--branch macos10-15` when creating the Vapor project - see [here](https://docs.vapor.codes/4.0/hello-world/#:~:text=Vapor%20and%20the%20template%20now%20uses%20async/await%20by%20default.%20If%20you%20cannot%20update%20to%20macOS%2012%20and/or%20need%20to%20continue%20to%20use%20EventLoopFutures%2C%20use%20flag%20%2D%2Dbranch%20macos10%2D15.)

**Review by zognin — APPROVED** [review]
LGTM for now

**Review by ganhongyao — APPROVED** [review]
LGTM

**Pull request merged**